index.js

```diff
@@ -1,5 +1,67 @@
 'use strict';
 
+var superagent = require('superagent');
+
+global.D2LAccessTokenExpiresAt = 0;
+
+function now() {
+	return Date.now()/1000 | 0;
+}
+
+function addHeaders(req) {
+	req.set('X-Csrf-Token', localStorage['XSRF.Token']);
+	req.set('X-D2L-App-Id', 'deprecated');
+}
+
+function processRefreshResponse(cb) { return function(res) {
+	if (!res.ok) {
+		return cb(false);
+	}
+
+	var cacheControl = res.headers['cache-control'];
+	if (!cacheControl) {
+		return cb(true);
+	}
+
+	var directives = cacheControl.split(',');
+	for (var i = 0; i < directives.length; i++) {
+		if (directives[i].indexOf('max-age') == -1) {
+			continue;
+		}
+
+		var maxAge = +directives[i].split('=')[1];
+		global.D2LAccessTokenExpiresAt = now() + maxAge;
+		break;
+	}
+
+	return cb(true);
+};}
+
+function refreshCookie(cb) {
+	var req = superagent
+		.post('/d2l/lp/auth/oauth2/refreshcookie');
+
+	addHeaders(req);
+
+	req.end(processRefreshResponse(cb));
```

Comment:
I'll have to confirm, but I think, rather than

```js
function processRefreshResponse(cb) {
  return function (res) {}
}
// ...
req.end(processRefreshResponse(cb));
```

The following would be more memory-friendly
```js
function processRefreshResponse (res, cb) {
}
// ...
req.end(function (res) { processRefreshReponse(res, cb); });
```

Given the function you were returning is a "larger"